base_layer/core/src/chain_storage/postgres_db/models/tx_outputs.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2020. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::{
+    blocks,
+    blocks::BlockHash,
+    chain_storage::{
+        postgres_db::{models::error::PostgresError, schema::*},
+        DbKeyValuePair,
+        DbValue,
+    },
+    transactions::types::{BlindingFactor, HashOutput},
+};
+use chrono::{NaiveDateTime, Utc};
+use diesel::{self, expression::dsl, prelude::*, OptionalExtension};
+use log::*;
+use serde_json::Value;
+use std::convert::{TryFrom, TryInto};
+use tari_crypto::tari_utilities::{hex::Hex, Hashable};
+
+const LOG_TARGET: &str = "b::c::storage::postgres:tx_outputs";
+
+#[derive(Queryable, Identifiable, Insertable)]
+#[table_name = "tx_outputs"]
+#[primary_key(hash)]
+pub struct TxOutput {
+    hash: String,
+    features_flags: i16,
+    features_maturity: i64,
+    commitment: String,
+    proof: Vec<u8>,
+    tx_output: String,
+    spent: String,
```

Comment:
```suggestion
    spent_in_block: String,
```

What is it when unspent? Isn't it better as a Option<String>?

Reply:
I changed it to option in the previous push.